_data-prepper/peer-forwarder.md

```diff
@@ -0,0 +1,183 @@
+| `aws_region` | A `String` represents the AWS region to use `ACM`, `S3` or `AWS Cloud Map`. Required if `use_acm_certificate_for_ssl` is set to `true` or `ssl_certificate_file` and `ssl_key_file` is `AWS S3` path or if `discovery_mode` is set to `aws_cloud_map`. |
+
+```yaml
+peer_forwarder:
+  ssl: true
+  ssl_certificate_file: "<cert-file-path>"
+  ssl_key_file: "<private-key-file-path>"
+```
+
+## Authentication
+
+This section describes optional authentication.
+
+`authentication`(optional) : A `Map` that enables mTLS. It can either be `mutual_tls` or `unauthenticated`. The default value is `unauthenticated`.
+
+```yaml
+peer_forwarder:
+  authentication:
+    mutual_tls:
+```
+
+## Metrics
+
+This section describes custom metrics capabilities, including timer, counter, and gauge.
+
+Core Peer Forwarder introduces the following custom metrics and all the metrics are prefixed by `core.peerForwarder`.
+
+### Timer
+
+This section describes the metrics feature, timer.
+
+- `requestForwardingLatency`: measures latency of forwarding requests by peer forwarder client.
+- `requestProcessingLatency`: measures latency of processing requests by peer forwarder server.
+
+### Counter
+
+See the table below for counter metric options.
+
+| Value | Description |
+| ----- | ----------- |
+| `requests`| measures total number of forwarded requests. |
+| `requestsFailed`| measures total number of failed requests. Requests with HTTP response code other than `200`. |
+| `requestsSuccessful`|  measures total number of successful requests. Requests with HTTP response code `200`. |
+| `requestsTooLarge`| measures total number of requests which are too large to be written to peer forwarder buffer. Requests with HTTP response code `413`. |
+| `requestTimeouts`| measures the total number of requests which timed out while writing content to peer forwarder buffer. Requests with HTTP response code `408`. |
+| `requestsUnprocessable`| measures total number of requests which failed due to unprocessable entity. Requests with HTTP response code `422`. |
+| `badRequests`| measures total number of requests with bad request format. Requests with HTTP response code `400`. |
+| `recordsSuccessfullyForwarded`| measures total number of forwarded records successfully. |
+| `recordsFailedForwarding`| measures total number of records failed to be forwarded. |
+| `recordsToBeForwarded` | measures total number of records to be forwarded. |
+| `recordsToBeProcessedLocally` | measures total number of records to be processed locally. |
+| `recordsActuallyProcessedLocally`| measures total number of records actually processed locally. Sum of  recordsToBeProcessedLocally` and `recordsFailedForwarding`. |
+| `recordsReceivedFromPeers`| measures total number of records received from remote peers. |
+
+### Gauge
+
+This section describes gauge metric options.
+
+`peerEndpoints` measures number of dynamically discovered peer data-prepper endpoints. For `static` mode, the size is fixed.
```

Comment:
Is "data-prepper"spelled this way intentionally? That is instead of "peer Data Prepper endpoints?"